### ISSUE TITLE
How to choose the best iter?

### ISSUE THREAD

**Issue by SherlockHolmes221** [DirtyHarryLYL/Transferable-Interactiveness-Network#37]
Thanks for your great job！
I want to know how to choose the best iter to test as I found the you didn't set the validation set in your code.

**Comment by DirtyHarryLYL** [comment]
In fact, we cannot know the exact 'best' iteration, just use the experience and the loss curve to choose an experiential iteration.
According to our experience, when using cosine lr decay, a good iteration can be found at about 3~5 complete annealing decay. Usually, we set the first decay as 2 epochs of the training set. More details can be found in the paper: SGDR: Stochastic Gradient Descent with Warm Restarts.

**Comment by SherlockHolmes221** [comment]
Thanks! 
What about training v-coco, I see that your just train 20000 iteration and the result of 6000 iteration  preforms a good result. However, when I training the vcoco dataset, I found the loss continued to go down when the iteration reached 20000 and the testing result was not good.

**Comment by DirtyHarryLYL** [comment]
A more detailed analysis of the classes may provide you more clues about the fitting state of the model, some classes' performances begin to degrade after xx iterations, meanwhile, some other classes are still increasing. V-COCO is quite smaller, it exactly needs much less iterations to achieve a good point and has a very early stopping in our experiment.

**Comment by SherlockHolmes221** [comment]
Thanks a lot!
